client/core/blockchain.go

```diff
@@ -902,21 +909,39 @@ func (bc *BlockChain) WriteBlockWithState(block *types.Block, receipts []*types.
 
 	rawdb.WriteReceipts(batch, block.Hash(), block.NumberU64(), receipts)
 
-	// Reorganise the chain if the parent is not the head block
-	if block.ParentHash() != currentBlock.Hash() {
-		if err := bc.reorg(currentBlock, block); err != nil {
-			return NonStatTy, err
+	//todo: @jeka try to use cumulative votes amount
```

Comment:
hard to tell what this is doing, can you use functions that describe these operations

ex: 
`isSameBlockHigh`
`isBlockHigher`
`needsReorg`
`isHeadBlock`
...